### PR TITLE
enable clippy for bootstrap on CI PRs (in `mingw-check` image)

### PR DESCRIPTION
**Pull request by onur-ozkan**
Let's keep the bootstrap codebase cleaner.

### CODE REVIEW

File: src/bootstrap/src/utils/helpers.rs [hunk 1]
```diff
@@ -549,7 +549,12 @@ pub fn hex_encode<T>(input: T) -> String
 where
     T: AsRef<[u8]>,
 {
-    input.as_ref().iter().map(|x| format!("{:02x}", x)).collect()
+    use std::fmt::Write;
+
+    input.as_ref().iter().fold(String::new(), |mut acc, &byte| {
```
Comment (albertlarsan68) [inline src/bootstrap/src/utils/helpers.rs:554]: Could we pre-allocate the string here?

**Review by albertlarsan68 — CHANGES_REQUESTED** [review]
Looks great overall, just a tiny nit.

File: src/bootstrap/src/utils/helpers.rs (hunk 1 above)
Comment (onur-ozkan) [reply src/bootstrap/src/utils/helpers.rs]: Good idea, thanks! (https://github.com/rust-lang/rust/compare/910eaae9048b243a247cfbf472d945a0c85757e8..c5fdb391208b309959a63f9798f85b0179bf947e).

**Review by albertlarsan68 — APPROVED** [review]
Thanks for the PR!

**Pull request merged**